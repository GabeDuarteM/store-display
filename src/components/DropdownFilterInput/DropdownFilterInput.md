```js
<DropdownFilterInput
  getButtonProps={customProps => customProps}
  handleChangeMenuIsOpen={() => {}}
  isOpen={false}
  selectedItem={{ id: 'price-asc', text: 'Price: low to high' }}
  width={200}
/>
```
