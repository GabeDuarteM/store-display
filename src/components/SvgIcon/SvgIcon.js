import React from 'react'

import PropTypes from 'prop-types'
import injectSheet from 'react-jss'
import cn from 'classnames'

export const styles = {
  svgIcon: {
    height: ({ height }) => height,
    width: ({ width }) => width,
    fill: ({ color }) => color,
    display: 'block',
  },
}

// eslint-disable-next-line no-unused-vars
const SvgIcon = ({
  classes,
  src,
  height,
  width,
  color,
  className,
  ...rest
}) => (
  <span
    className={cn(classes.svgIcon, className)}
    // eslint-disable-next-line react/no-danger
    dangerouslySetInnerHTML={{ __html: src }}
    {...rest}
  />
)

export const propTypes = {
  /** @ignore */
  classes: PropTypes.object.isRequired,
  /** If passed, the className will be applied to the root of the component */
  className: PropTypes.string,
  /** A string containing the content of the svg */
  src: PropTypes.string.isRequired,
  /** The height of the svg */
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** The width of the svg */
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** The color to use as the fill of the svg */
  color: PropTypes.string,
}
export const defaultProps = {
  height: 'auto',
  width: 'auto',
  color: null,
  className: null,
}

SvgIcon.propTypes = propTypes
SvgIcon.defaultProps = defaultProps

export default injectSheet(styles)(SvgIcon)
