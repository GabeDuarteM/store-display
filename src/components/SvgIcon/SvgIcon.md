```js
const downArrow = require('../../assets/down-arrow.svg')
;<SvgIcon width={10} src={downArrow} />
```
