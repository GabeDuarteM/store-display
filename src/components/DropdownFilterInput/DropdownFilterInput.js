import React from 'react'

import PropTypes from 'prop-types'
import cn from 'classnames'
import injectSheet from 'react-jss'

import downArrow from '../../assets/down-arrow.svg'
import SvgIcon from '../SvgIcon'

export const styles = {
  dropdownFilter: {
    WebkitAppearance: 'none !important',
    boxSizing: 'border-box',
    height: 36,
    paddingLeft: 15,
    width: ({ width }) => width,
    alignItems: 'center',
    display: 'flex',
    fontSize: 12,
    border: ({ isOpen }) => `1px solid #${isOpen ? '6e8be3' : 'd7d8db'}`,
    borderRadius: 3,
    color: '#585858',
    backgroundColor: 'white',
    '&:active': {
      border: '1px solid #6e8be3',
    },
  },
  downArrow: {
    marginLeft: 'auto',
    paddingRight: 16,
    '&.isOpen': {
      transform: 'rotateX(180deg)',
    },
  },
}

const DropdownFilterInput = ({
  classes,
  getButtonProps,
  handleChangeMenuIsOpen,
  selectedItem,
  isOpen,
}) => (
  <div
    role="button"
    {...getButtonProps({
      onClick: () => {
        handleChangeMenuIsOpen(!isOpen)
      },
      className: classes.dropdownFilter,
    })}
  >
    {selectedItem.text}
    <SvgIcon
      className={cn(classes.downArrow, { isOpen })}
      width={9}
      color="#9B9B9B"
      src={downArrow}
    />
  </div>
)

export const propTypes = {
  /** @ignore */
  classes: PropTypes.object.isRequired,
  /** Function to get the props from Downshift */
  getButtonProps: PropTypes.func.isRequired,
  /** Property that define if the item list is opened or not */
  isOpen: PropTypes.bool.isRequired,
  /** Gets called when the user clicks the input. It used to open the menu with the items. */
  handleChangeMenuIsOpen: PropTypes.func.isRequired,
  /** The current selected item to be displayed on the input */
  selectedItem: PropTypes.shape({
    id: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
  }).isRequired,
  /** The width of the component */
  width: PropTypes.number.isRequired,
}

DropdownFilterInput.propTypes = propTypes

export default injectSheet(styles)(DropdownFilterInput)
