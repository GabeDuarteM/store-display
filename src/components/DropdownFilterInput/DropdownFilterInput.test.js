import React from 'react'

import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import DropdownFilterInput, { styles, propTypes } from './DropdownFilterInput'
import SvgIcon from '../SvgIcon'

const createWrapper = (func, optionalProps) =>
  func(
    <DropdownFilterInput
      getButtonProps={jest.fn(customProps => customProps)}
      handleChangeMenuIsOpen={jest.fn()}
      isOpen
      selectedItem={{ id: '', text: 'selectedItem' }}
      width={123}
      {...optionalProps}
    />,
  )

const createWrapperShallow = optionalProps =>
  createWrapper(shallow, optionalProps)

const createWrapperMount = optionalProps => createWrapper(mount, optionalProps)

describe('COMPONENT: <DropdownFilterInput />', () => {
  it('should render without crashing with default props', () => {
    createWrapperMount()
  })

  it("should match component's snapshot", () => {
    const wrapper = createWrapperShallow()
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

  it("should match component's style snapshot", () => {
    expect(styles).toMatchSnapshot()
  })

  it("should match component's propTypes snapshot", () => {
    expect(propTypes).toMatchSnapshot()
  })

  it('should spread the return of the method getButtonProps on the root component', () => {
    const wrapper = createWrapperShallow()
    const { onClick, className } = wrapper.dive().props()
    expect(onClick).toBeDefined()
    expect(className).toBeDefined()
  })

  it('should call handleChangeMenuIsOpen passing the opposite of the isOpen state when the root component is clicked', () => {
    const handleChangeMenuIsOpen = jest.fn()
    const isOpen = false
    const wrapper = createWrapperShallow({ handleChangeMenuIsOpen, isOpen })
    wrapper.dive().simulate('click')
    expect(handleChangeMenuIsOpen).toHaveBeenCalledTimes(1)
    expect(handleChangeMenuIsOpen).toHaveBeenLastCalledWith(!isOpen)
  })

  it('should render the selected item inside the root component', () => {
    const selectedItem = { id: 'price', text: 'Price' }
    const wrapper = createWrapperShallow({ selectedItem })

    expect(wrapper.render().text()).toBe(selectedItem.text)
  })

  it('should render a SvgIcon inside the root component', () => {
    const wrapper = createWrapperShallow()

    expect(
      wrapper
        .dive()
        .find(SvgIcon)
        .exists(),
    ).toBeTruthy()
  })

  it('should render a SvgIcon with an isOpen className when the prop isOpen is true', () => {
    const wrapper = createWrapperShallow({ isOpen: true })

    expect(
      wrapper
        .dive()
        .find(SvgIcon)
        .hasClass('isOpen'),
    ).toBeTruthy()
  })

  it('should render a SvgIcon without an isOpen className when the prop isOpen is false', () => {
    const wrapper = createWrapperShallow({ isOpen: false })

    expect(
      wrapper
        .dive()
        .find(SvgIcon)
        .hasClass('isOpen'),
    ).toBeFalsy()
  })
})
