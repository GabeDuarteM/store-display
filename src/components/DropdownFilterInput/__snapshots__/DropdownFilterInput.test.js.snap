// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT: <DropdownFilterInput /> should match component's propTypes snapshot 1`] = `
Object {
  "classes": [Function],
  "getButtonProps": [Function],
  "handleChangeMenuIsOpen": [Function],
  "isOpen": [Function],
  "selectedItem": [Function],
  "width": [Function],
}
`;

exports[`COMPONENT: <DropdownFilterInput /> should match component's snapshot 1`] = `
<div
  className="DropdownFilterInput-dropdownFilter DropdownFilterInput-dropdownFilter"
  onClick={[Function]}
  role="button"
>
  selectedItem
  <Jss(SvgIcon)
    className="DropdownFilterInput-downArrow DropdownFilterInput-downArrow isOpen"
    color="#9B9B9B"
    height="auto"
    src=""
    width={9}
  />
</div>
`;

exports[`COMPONENT: <DropdownFilterInput /> should match component's style snapshot 1`] = `
Object {
  "downArrow": Object {
    "&.isOpen": Object {
      "transform": "rotateX(180deg)",
    },
    "marginLeft": "auto",
    "paddingRight": 16,
  },
  "dropdownFilter": Object {
    "&:active": Object {
      "border": "1px solid #6e8be3",
    },
    "WebkitAppearance": "none !important",
    "alignItems": "center",
    "backgroundColor": "white",
    "border": [Function],
    "borderRadius": 3,
    "boxSizing": "border-box",
    "color": "#585858",
    "display": "flex",
    "fontSize": 12,
    "height": 36,
    "paddingLeft": 15,
    "width": [Function],
  },
}
`;
